### PR TITLE
Increase timeout for docs PR link action

### PR DESCRIPTION
**Pull request by robbavey**
Update the timeout from 15 minutes to 30 minutes to try and fix the inline docs previews feature

### CODE REVIEW

File: .github/workflows/add-docs-preview-link.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
           statusName: "buildkite/docs-build-pr"
           # https://elasticsearch-ci.elastic.co/job/elastic+logstash+pull-request+build-docs
           # usually finishes in ~ 20 minutes
-          timeoutSeconds: 900
+          timeoutSeconds: 1800
```
Comment (donoghuc) [inline .github/workflows/add-docs-preview-link.yml:25]: The comment and original value dont really make sense? If it typically finishes in ~20 minutes, why would the timeout be 15 minutes? Also, the URI in the comment does not seem to be correct either. I'm assuming this is somehow tracking the buildkite job? The github action history make it look like this just always goes for about 15 minutes (the timeout threshold) but it looks like buildkite jobs can finish much fater than that https://buildkite.com/elastic/docs-build-pr/builds/194330 for example?

Comment (donoghuc) [reply .github/workflows/add-docs-preview-link.yml:25]: I think the root of my misunderstanding here is what this is meant to track and how to validate that status https://github.com/elastic/logstash/blob/eb7e1253e003a09e3297321ec760ba73193d17b8/.github/workflows/add-docs-preview-link.yml#L22

Comment (robbavey) [reply .github/workflows/add-docs-preview-link.yml:25]: Let me clear up the comments - there are a bunch of mixed messages here from previous commits. 

The builds take ~ 20 minutes, so the 30 minute timeout should be sufficient.

It took me a little digging to find the [right incantation](https://github.com/elastic/docs/blob/ca7de98a85cf43a544a5e3551807d2c928028902/.github/actions/docs-preview/action.yml#L32C35-L32C118) to find product specific builds - the latest one for example is https://buildkite.com/elastic/docs-build-pr/builds?created_from_days=7&meta_data%5Brepo_pr%5D=logstash_16701

**Review by donoghuc — APPROVED** [review]
I think I see why I have been quite understanding this. The [triggers](https://github.com/elastic/logstash/blob/eb7e1253e003a09e3297321ec760ba73193d17b8/.github/workflows/add-docs-preview-link.yml#L5) are on `Open` and `Synchronize` **and** have some path constraints. Seems like this GH action ran on open https://github.com/elastic/logstash/actions/runs/11961243472/job/33347029004 and timed out waiting for https://buildkite.com/elastic/docs-build-pr/builds/194279 (which took 21 mins). But have not been running on synchronize (you pushing another commit for example) because the path constraint is not satisfied. 

I dont quite see how the name `buildkite/docs-build-pr` is generated or managed, but given that is what displays in the UI i'm sure that is right.

**Pull request merged**